grafana/google-sheets-datasource
# Fix timeout errors

Follow up on https://github.com/grafana/google-sheets-datasource/pull/253 to also include timeout errors.

Fixed error messages (currently these are plugin error):

```
ERROR[08-06|16:01:08] Partial data response error              logger=plugin.metrics refID=A status=504 error="context deadline exceeded (Client.Timeout or context cancellation while reading body)" statusSource=downstream
ERROR[08-06|16:01:08] Plugin Request Completed                 logger=plugin.metrics status=error duration=8.456452958s pluginId=grafana-googlesheets-datasource endpoint=queryData statusSource=downstream uname=admin stackId=42 slug=unknownSlug dsName=cold-plunges dsUid=e6ea1a43-c46a-4d27-9a97-afaabe4c91f4
```

How to test: 
In `createSheetsService` in `pkg/googlesheets/googleclient.go`  add `client.Timeout = 1 * 1000 * 1000` to line 139 (under where we create a client). This will create timeout error that should be marked as downstream.
